### PR TITLE
DBEX: make pdf extension evaluation case insensitive

### PR DESCRIPTION
**Pull request by freeheeling**
This PR fixes an issue where password-protected PDFs whose extensions were not downcased were uploaded to S3 without first being unlocked, thus resulting in the documents being inaccessible to downstream services.

## Summary

- This work is behind a feature toggle (flipper): ~YES~/NO
- In addition to making the pdf extension evaluation case insensitive, the RegEx utilized for sanitizing error messages has also been modified to match the extension pattern case-insensitively.
- Disability Benefits Experience Team 2 (dBeX Carbs 🥖)

## Related issue(s)

- [78539](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/va.gov-team/78539)

## Testing done

- Corresponding test file passes with extension casing change

## What areas of the site does it impact?

526EZ application document uploads and other forms that accept password-protected PDFs

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: app/models/form_attachment.rb [hunk 1]
```diff
@@ -36,15 +36,15 @@ def get_file
   private
 
   def unlock_pdf(file, file_password)
-    return file unless File.extname(file) == '.pdf'
+    return file unless File.extname(file).downcase == '.pdf'
```
Comment (shaunburdick) [inline app/models/form_attachment.rb:39]: What if this line is removed all together?
With a method name of `unlock_pdf`, it should already be determined that it's a PDF before it's called. 
Ideally, it should be the responsibility of the caller to make sure it's a PDF before calling this method. (That way they can do a better job than checking the file extension...)

Comment (shaunburdick) [reply app/models/form_attachment.rb:39]: You could move the type check up to line 16 where I think it makes more sense, logically.

Comment (freeheeling) [reply app/models/form_attachment.rb]: Relocated extension type check per suggestion. Thanks, @shaunburdick.

**Review by shaunburdick — APPROVED** [review]
LGTM!

**Review by ericboehs — APPROVED** [review]

**Pull request merged**